### PR TITLE
Fixed based on proxy issue #292

### PR DESCRIPTION
**Pull request by amp5208**
As mentioned by user @oshingc, there is a bug in the proxy / auth proxy string validation with the Jira collector that needs to be resolved.  If a user has not set a proxy, but their property file has the proxy / auth proxy props set as blank, or not there at all, the tool will currently throw a URI syntax or null pointer exception, as indicated in the following issue:  https://github.com/capitalone/Hygieia/issues/292

This has now been fixed.  Please validate and merge as soon as possible.  I have also wrote additional test cases to cover these scenarios for Jira connections.

### CODE REVIEW

**Comment by amp5208** [comment]
I am not using the URI constructors prior to my local constructor to support backwards compatibility.  If you look at my constructor, however, I am, in fact, directly leveraging the URI constructors.

StringUtils.isEmpty() did not work when I created this fix in this scenario for some reason.  However, let me try again and I will push an update.

File: jira-feature-collector/src/main/java/com/capitalone/dashboard/datafactory/jira/JiraDataFactoryImpl.java [hunk 1]
```diff
@@ -88,9 +89,21 @@ public JiraDataFactoryImpl(String jiraCredentials, String jiraBaseUrl) {
 	 */
 	public JiraDataFactoryImpl(String jiraCredentials, String jiraBaseUrl, String jiraProxyUrl,
 			String jiraProxyPort) {
-		URI jiraUri = this.createJiraConnection(jiraBaseUrl, jiraProxyUrl + ":" + jiraProxyPort,
-				this.decodeCredentials(jiraCredentials).get("username"),
-				this.decodeCredentials(jiraCredentials).get("password"));
+		URI jiraUri;
+		if ((jiraProxyUrl != null) && (jiraProxyPort != null) && (jiraProxyUrl != BLANK_LITERAL)
```
Comment (jdamick) [inline jira-feature-collector/src/main/java/com/capitalone/dashboard/datafactory/jira/JiraDataFactoryImpl.java]: just use Strings.isNullOrEmpty()

File: jira-feature-collector/src/main/java/com/capitalone/dashboard/datafactory/jira/JiraDataFactoryImpl.java [hunk 2]
```diff
@@ -145,9 +158,21 @@ public JiraDataFactoryImpl(int inPageSize, String jiraCredentials, String jiraBa
...
+		} else {
+			try {
+				LOGGER.debug("Handling without authenticated proxy (fields were available in properties settings but were blank)");
+				jiraUri = new URI(jiraBaseUrl);
+			} catch (URISyntaxException e) {
+				LOGGER.error("There was a problem reading the provide Jira base URI syntax");
+				jiraUri = null;
+			}
+		}
 		this.client = factory.createWithBasicHttpAuthentication(jiraUri,
```
Comment (jdamick) [inline jira-feature-collector/src/main/java/com/capitalone/dashboard/datafactory/jira/JiraDataFactoryImpl.java]: if jiraUri is null why do this?

**Comment by amp5208** [comment]
I will switch to .isNullOrEmpty()

That section runs no matter what the user sets in their property file, and is run by the local clients.  For this reason, I have to validate all scenarios.  Did you have a suggestion on a better implementation here?

**Comment by oshingc** [comment]
!= null definitely catch the case of a property with no value I tested that when proxyUrl was empty.
For now those validations are ok for solve the issue, 
Thanks @amp5208

**Comment by amp5208** [comment]
Thank you @oshingc.  I have just pushed an update with the style suggestions from @tabladrum and @jdamick  above.  I also included additional connectivity checks for Jira.  Please validate once more for your configuration, if you are able.  Thank you!

**Comment by oshingc** [comment]
It's working @amp5208

File: jira-feature-collector/src/main/java/com/capitalone/dashboard/datafactory/jira/JiraDataFactoryImpl.java [hunk 3]
```diff
@@ -145,12 +162,29 @@ public JiraDataFactoryImpl(int inPageSize, String jiraCredentials, String jiraBa
...
 	public JiraDataFactoryImpl(int inPageSize, String jiraCredentials, String jiraBaseUrl,
 			String jiraProxyUrl, String jiraProxyPort) {
-		URI jiraUri = this.createJiraConnection(jiraBaseUrl, jiraProxyUrl + ":" + jiraProxyPort,
-				this.decodeCredentials(jiraCredentials).get("username"),
-				this.decodeCredentials(jiraCredentials).get("password"));
-		this.client = factory.createWithBasicHttpAuthentication(jiraUri,
-				this.decodeCredentials(jiraCredentials).get("username"),
-				this.decodeCredentials(jiraCredentials).get("password"));
+		URI jiraUri;
+		if ((jiraBaseUrl != null) && (!jiraBaseUrl.isEmpty())) {
```
Comment (jdamick) [inline jira-feature-collector/src/main/java/com/capitalone/dashboard/datafactory/jira/JiraDataFactoryImpl.java]: can just Strings.isNullOrEmpty()  - we're pulling in guava, might as well use it..

File: jira-feature-collector/src/main/java/com/capitalone/dashboard/datafactory/jira/JiraDataFactoryImpl.java [hunk 4]
```diff
@@ -145,12 +162,29 @@ public JiraDataFactoryImpl(int inPageSize, String jiraCredentials, String jiraBa
...
+			} else {
+				try {
+					LOGGER.debug("Handling without authenticated proxy (fields were available in properties settings but were blank)");
+					jiraUri = new URI(jiraBaseUrl);
+				} catch (URISyntaxException e) {
+					LOGGER.error("There was a problem reading the provide Jira base URI syntax");
+					jiraUri = null;
+				}
+			}
+			this.client = factory.createWithBasicHttpAuthentication(jiraUri, this
```
Comment (jdamick) [inline jira-feature-collector/src/main/java/com/capitalone/dashboard/datafactory/jira/JiraDataFactoryImpl.java]: so if the uri is null why bother try, should probably just throw?

**Comment by amp5208** [comment]
That's true I suppose, but PMD did not print any violations to fail the CI checks.  @jdamick , is it okay if I leave the extra parenthesis in my logic statements for the last push?

That's great to hear, and thank you for closing the ticket.  @jdamick and @tabladrum :  Once you are good with code style, you can merge this pull.

**Comment by amp5208** [comment]
I was avoiding throwing null pointers and rather handle... but let me see what I can do.

**Comment by amp5208** [comment]
Updated based on suggestions by @tabladrum and @jdamick .  Pending review.  Tests pass and all cases set before this commit should work the same.

**Comment by amp5208** [comment]
I have also pushed an important fix related to jira connection timeouts under production load.  Please accept this pull request to also get this fix.
